src/pymd4c.c

```diff
@@ -252,11 +252,106 @@ typedef struct {
 static int GenericParser_init(GenericParserObject *self, PyObject *args,
         PyObject *kwds) {
     unsigned int parser_flags = 0;
+    unsigned short int collapse_whitespace = 0;
+    unsigned short int permissive_atx_headers = 0;
+    unsigned short int permissive_url_autolinks = 0;
+    unsigned short int permissive_email_autolinks = 0;
+    unsigned short int no_indented_code_blocks = 0;
+    unsigned short int no_html_blocks = 0;
+    unsigned short int no_html_spans = 0;
+    unsigned short int tables = 0;
+    unsigned short int strikethrough = 0;
+    unsigned short int permissive_www_autolinks = 0;
+    unsigned short int tasklists = 0;
+    unsigned short int latex_math_spans = 0;
+    unsigned short int wikilinks = 0;
+    unsigned short int underline = 0;
     
-    static char *kwlist[] = {"parser_flags", NULL};
-    if (!PyArg_ParseTupleAndKeywords(args, kwds, "|I", kwlist, &parser_flags)) {
+    static char *kwlist[] = {
+        "parser_flags",
+        "collapse_whitespace",
+        "permissive_atx_headers",
+        "permissive_url_autolinks",
+        "permissive_email_autolinks",
+        "no_indented_code_blocks",
+        "no_html_blocks",
+        "no_html_spans",
+        "tables",
+        "strikethrough",
+        "permissive_www_autolinks",
+        "tasklists",
+        "latex_math_spans",
+        "wikilinks",
+        "underline",
+        NULL
+    };
+    if (!PyArg_ParseTupleAndKeywords(args, kwds, "|I$pppppppppppppp", kwlist,
+                                     &parser_flags, &collapse_whitespace,
+                                     &permissive_atx_headers,
+                                     &permissive_url_autolinks,
+                                     &permissive_email_autolinks,
+                                     &no_indented_code_blocks, &no_html_blocks,
+                                     &no_html_spans, &tables, &strikethrough,
+                                     &permissive_www_autolinks, &tasklists,
+                                     &latex_math_spans, &wikilinks, &underline)) {
```

Comment:
According to the [C API](https://docs.python.org/3/c-api/arg.html#other-objects), the `p` format expects to write to an `int`, but we are providing pointers to `unsigned short int`. I suspect if you added a test where two adjacent flags were set, there's a decent chance one of them would end up unset. (If not, we are just lucky.)

The space savings would be nice, but I think these variables all need to be plain ol' signed `int`.